api/api/specs/routers.yaml

```diff
@@ -580,6 +582,18 @@ components:
         - "pending"
       default: "pending"
 
+    RouteProtocol:
```

Comment:
I think `RouteProtocol` vs `RouterProtocol` will be confusing to differentiate, can we find better name? And also, we should differentiate `protocol` field in `RouterConfig` and the `protocol` field in `Route` since both are called `protocol` but accept different type.

Reply:
how about changing `RouterProtocol` to `RouterProtocolType` and `router_protocol_type` in json, since its more router specific as well? 

While keeping the one for `Route` same since `endpoint`, `protocol`, `service_method` seems to more generic.

Reply:
Maybe `RouterType` ? So we have `HTTP` router and `GRPC` router.

Reply:
I thought of that but it could refer to many different things without context. No strong opinion either way

Reply:
on second thought, while keeping Route Protocol intact to be in sync with Fiber struct but changing `RouterProtocol` to `RouterType` will be confusing since Merlin is using `Protocol` for http/upi/grpc, which is identical to `RouterType` but not `Protocol` of Fiber (GRPC/HTTP only, no UPI).

Changing `Protocol` of Fiber to `RouteType` is confusing as well, as there is already a `type` field to indicate the type of Fiber component. 

Fiber.Protocol = HTTP/GRPC
Merlin.Protocol = Router.Protocol = HTTP/GRPC/UPI

In short, if we sync with Merlin, it will clash with Fiber since both `Protocol` means different thing in context.

What I can propose is to change Fiber and Routes (although `protocol` make total sense within that context) to AppProtocol.

For router, to sync with merlin
Protocol RouterProtocol `json:"protocol"` -> Protocol Protocol `json:"protocol"` 
For Fiber and Routes (turing model)
Protocol RouteProtocol `json:"protocol"` -> AppProtocol AppProtocol `json:"app_protocol"` 
Fiber to rename `Protocol` type to `AppProtocol`, likewise throughout the config.

wdyt?

Reply:
Ah I see.

I agree with the first part of the proposal. Let's make it consistent with Merlin fo the `RouterProtocol`.

On the other hand,  for fiber protocol

> Protocol RouteProtocol json:"protocol" -> AppProtocol AppProtocol json:"app_protocol"
Fiber to rename Protocol type to AppProtocol, likewise throughout the config.

Does it make sense to call it `ServiceType` ? so we have `HTTP` service and `GRPC` service at fiber level? Other alternatives: `AppType`, `RouteProtocol`.

Reply:
asked for a couple of opinions, `RouteProtocol` still made the most sense. Although having `route` appended looks weird it itself, cause usually for naming, we do `animal.name` instead of `animal.animal_name`. Even if two object has the same field, there is never a guarantee for them to be the same type/struct. Either way to me its just different kind of confusion, regardless user should always follow the swagger doc and input accordingly 😅 .

If no more opinion will change to this 😅 
For router, to sync with merlin, `config.Protocol` is define in router code [upi or http].
`Protocol RouterProtocol json:"protocol"` -> `Protocol config.Protocol json:"protocol"`
For Routes (turing model) 
`Protocol RouteProtocol json:"protocol"` -> `RouteProtocol RouteProtocol json:"route_protocol"`
Fiber will continue to use `protocol` in config and code (looks really weird if its `request.route_protocol()` instead of `request.protocol()` )

Reply:
Do we need to store the route protocol? If we assume that protocols will not be mixed, can we infer the route protocol from that of the router at deploy time (HTTP_JSON -> http, UPI_V1 / CUSTOM_GRPC -> grpc)?

Reply:
I thought of that but decided to add it to the model for it to be 100% reflective of the fiber.yaml's route section. We could drop it (removes confusion with protocol) as well actually, now that i think of it.